src/main/scala/hellotest/FunctionalWordCloud.scala

```diff
@@ -59,4 +59,20 @@ def sortCount(map: Map[String,Int]): Iterator[(String,Int)] =
 def convert(wordCount: Iterator[(String, Int)]): String = {
   // DO: Take in iterator of word counts and convert into a string (ex: Iterator(("bb",2),("aa",2)) -> "bb: 2 aa: 2")
   wordCount.map { case (word, count) => s"$word: $count" }.mkString(" ")
+}
+
+def getValidLengthSequences(sequences: Iterator[Seq[String]], minLength: Int): Iterator[Seq[String]] = {
+  sequences.filter(seq => seq.length >= minLength)
+}
+
+def mapWordCounts(sequences: Iterator[Seq[String]], cloudSize: Int): Iterator[Map[String, Int]] = {
+  sequences.map(sequence => countFrequencies(sequence, cloudSize))
+}
+
+def sortWordCounts(wordCounts: Iterator[Map[String, Int]]): Iterator[(String, Int)] = {
+  wordCounts.flatMap(sortCount)
```

Comment:
Don't use `flatMap`. The return result should be an iterator of iterators
Ex:
```
Iterator(
  Iterator(("bb",2),("aa",1)),
  Iterator(("aa",2),("bb",1))
)
```
Right now it's turning everything into this:
```
Iterator(
  ("bb",2),
  ("aa",1),
  ("aa",2),
  ("bb",1)
)
```
That's why I had the 
```
wordCounts.map( _ => sortWordCounts(_))
```
in the run function so it wouldn't flatten the sequences